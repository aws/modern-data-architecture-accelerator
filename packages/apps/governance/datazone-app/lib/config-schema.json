{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "additionalProperties": false,
    "definitions": {
        "AdminUser": {
            "additionalProperties": false,
            "properties": {
                "role": {
                    "$ref": "#/definitions/MdaaRoleRef"
                },
                "userType": {
                    "enum": [
                        "IAM_ROLE",
                        "SSO_USER"
                    ],
                    "type": "string"
                }
            },
            "required": [
                "role",
                "userType"
            ],
            "type": "object"
        },
        "AsscociatedAccount": {
            "additionalProperties": false,
            "properties": {
                "account": {
                    "type": "string"
                },
                "adminUsers": {
                    "$ref": "#/definitions/NamedAdminUsers"
                },
                "cdkRoleArn": {
                    "type": "string"
                },
                "region": {
                    "type": "string"
                }
            },
            "required": [
                "account"
            ],
            "type": "object"
        },
        "CfnParameterProps": {
            "additionalProperties": false,
            "properties": {
                "allowedPattern": {
                    "default": "- No constraints on patterns allowed for parameter.",
                    "description": "A regular expression that represents the patterns to allow for String types.",
                    "type": "string"
                },
                "allowedValues": {
                    "default": "- No constraints on values allowed for parameter.",
                    "description": "An array containing the list of values allowed for the parameter.",
                    "items": {
                        "type": "string"
                    },
                    "type": "array"
                },
                "constraintDescription": {
                    "default": "- No description with customized error message when user specifies invalid values.",
                    "description": "A string that explains a constraint when the constraint is violated.\nFor example, without a constraint description, a parameter that has an allowed\npattern of [A-Za-z0-9]+ displays the following error message when the user specifies\nan invalid value:",
                    "type": "string"
                },
                "default": {
                    "default": "- No default value for parameter.",
                    "description": "A value of the appropriate type for the template to use if no value is specified\nwhen a stack is created. If you define constraints for the parameter, you must specify\na value that adheres to those constraints."
                },
                "description": {
                    "default": "- No description for the parameter.",
                    "description": "A string of up to 4000 characters that describes the parameter.",
                    "type": "string"
                },
                "maxLength": {
                    "default": "- None.",
                    "description": "An integer value that determines the largest number of characters you want to allow for String types.",
                    "type": "number"
                },
                "maxValue": {
                    "default": "- None.",
                    "description": "A numeric value that determines the largest numeric value you want to allow for Number types.",
                    "type": "number"
                },
                "minLength": {
                    "default": "- None.",
                    "description": "An integer value that determines the smallest number of characters you want to allow for String types.",
                    "type": "number"
                },
                "minValue": {
                    "default": "- None.",
                    "description": "A numeric value that determines the smallest numeric value you want to allow for Number types.",
                    "type": "number"
                },
                "noEcho": {
                    "default": "- Parameter values are not masked.",
                    "description": "Whether to mask the parameter value when anyone makes a call that describes the stack.\nIf you set the value to ``true``, the parameter value is masked with asterisks (``*****``).",
                    "type": "boolean"
                },
                "type": {
                    "default": "String",
                    "description": "The data type for the parameter (DataType).",
                    "type": "string"
                }
            },
            "type": "object"
        },
        "DomainProps": {
            "additionalProperties": false,
            "properties": {
                "additionalAdminUsers": {
                    "$ref": "#/definitions/NamedAdminUsers"
                },
                "associatedAccounts": {
                    "$ref": "#/definitions/NamedAssociatedAccounts"
                },
                "dataAdminRole": {
                    "$ref": "#/definitions/MdaaRoleRef"
                },
                "description": {
                    "type": "string"
                },
                "domainVersion": {
                    "enum": [
                        "V1",
                        "V2"
                    ],
                    "type": "string"
                },
                "sagemakerUnifiedDomain": {
                    "type": "boolean"
                },
                "singleSignOnType": {
                    "enum": [
                        "DISABLED",
                        "IAM_IDC"
                    ],
                    "type": "string"
                },
                "userAssignment": {
                    "enum": [
                        "AUTOMATIC",
                        "MANUAL"
                    ],
                    "type": "string"
                }
            },
            "required": [
                "dataAdminRole",
                "singleSignOnType",
                "userAssignment"
            ],
            "type": "object"
        },
        "MdaaNagSuppressionByPath": {
            "additionalProperties": false,
            "properties": {
                "path": {
                    "type": "string"
                },
                "suppressions": {
                    "items": {
                        "additionalProperties": false,
                        "properties": {
                            "id": {
                                "type": "string"
                            },
                            "reason": {
                                "type": "string"
                            }
                        },
                        "required": [
                            "id",
                            "reason"
                        ],
                        "type": "object"
                    },
                    "type": "array"
                }
            },
            "required": [
                "path",
                "suppressions"
            ],
            "type": "object"
        },
        "MdaaNagSuppressionConfigs": {
            "additionalProperties": false,
            "properties": {
                "by_path": {
                    "items": {
                        "$ref": "#/definitions/MdaaNagSuppressionByPath"
                    },
                    "type": "array"
                }
            },
            "required": [
                "by_path"
            ],
            "type": "object"
        },
        "MdaaRoleRef": {
            "additionalProperties": false,
            "description": "A reference to an IAM role. Roles can be referenced by name, arn, and/or id.",
            "properties": {
                "arn": {
                    "description": "Reference role by arn",
                    "type": "string"
                },
                "id": {
                    "description": "Reference role by id",
                    "type": "string"
                },
                "immutable": {
                    "description": "Indicates whether the role should be considered immutable (defaults false)",
                    "type": "boolean"
                },
                "name": {
                    "description": "Reference role by name",
                    "type": "string"
                },
                "refId": {
                    "description": "A string which uniquely identifies the MdaaRoleRef within a scope.",
                    "type": "string"
                },
                "sso": {
                    "description": "If true, role name will be resolved to an SSO auto-generated role. Also implies immutability.",
                    "type": "boolean"
                }
            },
            "type": "object"
        },
        "MdaaServiceCatalogConstraintConfig": {
            "additionalProperties": false,
            "properties": {
                "description": {
                    "type": "string"
                },
                "rules": {
                    "additionalProperties": {
                        "$ref": "#/definitions/MdaaServiceCatalogConstraintRuleConfig"
                    },
                    "type": "object"
                }
            },
            "required": [
                "description",
                "rules"
            ],
            "type": "object"
        },
        "MdaaServiceCatalogConstraintRuleAssertionConfig": {
            "additionalProperties": false,
            "properties": {
                "assert": {
                    "type": "string"
                },
                "description": {
                    "type": "string"
                }
            },
            "required": [
                "assert",
                "description"
            ],
            "type": "object"
        },
        "MdaaServiceCatalogConstraintRuleCondititionConfig": {
            "additionalProperties": false,
            "type": "object"
        },
        "MdaaServiceCatalogConstraintRuleConfig": {
            "additionalProperties": false,
            "properties": {
                "assertions": {
                    "items": {
                        "$ref": "#/definitions/MdaaServiceCatalogConstraintRuleAssertionConfig"
                    },
                    "type": "array"
                },
                "condition": {
                    "$ref": "#/definitions/MdaaServiceCatalogConstraintRuleCondititionConfig"
                }
            },
            "required": [
                "assertions",
                "condition"
            ],
            "type": "object"
        },
        "MdaaServiceCatalogParameterConfig": {
            "additionalProperties": false,
            "properties": {
                "constraints": {
                    "$ref": "#/definitions/MdaaServiceCatalogConstraintConfig"
                },
                "props": {
                    "$ref": "#/definitions/CfnParameterProps"
                }
            },
            "required": [
                "props"
            ],
            "type": "object"
        },
        "MdaaServiceCatalogProductConfig": {
            "additionalProperties": false,
            "properties": {
                "launch_role_name": {
                    "type": "string"
                },
                "name": {
                    "type": "string"
                },
                "owner": {
                    "type": "string"
                },
                "parameters": {
                    "additionalProperties": {
                        "$ref": "#/definitions/MdaaServiceCatalogParameterConfig"
                    },
                    "type": "object"
                },
                "portfolio_arn": {
                    "type": "string"
                }
            },
            "required": [
                "name",
                "owner",
                "portfolio_arn"
            ],
            "type": "object"
        },
        "NamedAdminUsers": {
            "additionalProperties": {
                "$ref": "#/definitions/AdminUser"
            },
            "type": "object"
        },
        "NamedAssociatedAccounts": {
            "additionalProperties": {
                "$ref": "#/definitions/AsscociatedAccount"
            },
            "type": "object"
        },
        "NamedDomainsProps": {
            "additionalProperties": {
                "$ref": "#/definitions/DomainProps"
            },
            "type": "object"
        }
    },
    "properties": {
        "domains": {
            "$ref": "#/definitions/NamedDomainsProps"
        },
        "nag_suppressions": {
            "$ref": "#/definitions/MdaaNagSuppressionConfigs",
            "description": "Nag suppressions"
        },
        "service_catalog_product_config": {
            "$ref": "#/definitions/MdaaServiceCatalogProductConfig",
            "description": "Service Catalog Config\nIf specified, the configured module will be deployed as a Service Catalog product instead of directly to the environment"
        }
    },
    "required": [
        "domains"
    ],
    "type": "object"
}

