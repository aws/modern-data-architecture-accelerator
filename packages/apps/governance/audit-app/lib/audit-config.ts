/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParser, MdaaAppConfigParserProps, MdaaBaseConfigContents } from '@aws-mdaa/app';
import { BucketInventoryProps } from '@aws-mdaa/audit-l3-construct';
import { MdaaRoleRef } from '@aws-mdaa/iam-role-helper';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';

import * as configSchema from './config-schema.json';

export interface AuditConfigContents extends MdaaBaseConfigContents {
  /**
   * List of refs to roles which will be provided read access to the audit logs
   */
  readRoles?: MdaaRoleRef[];
  /**
   * List of source accounts from which audit logs will be accepted
   */
  sourceAccounts?: string[];
  /**
   * List of source regions from which audit logs will be accepted
   */
  sourceRegions?: string[];
  /**
   * Specifies the S3 prefix where inventory data will be accepted on the audit bucket
   */
  inventoryPrefix?: string;
  /**
   * The list of expected inventories
   */
  inventories?: BucketInventoryProps[];
}

export class AuditConfigParser extends MdaaAppConfigParser<AuditConfigContents> {
  public readonly readRoleRefs: MdaaRoleRef[];
  public readonly sourceAccounts: string[];
  public readonly sourceRegions: string[];
  public readonly inventoryPrefix: string;
  public readonly inventories?: BucketInventoryProps[];

  constructor(stack: Stack, props: MdaaAppConfigParserProps) {
    super(stack, props, configSchema as Schema);

    this.readRoleRefs = this.configContents.readRoles ? this.configContents.readRoles : [];
    this.sourceAccounts = this.configContents.sourceAccounts ? this.configContents.sourceAccounts : [];
    this.sourceRegions = this.configContents.sourceRegions ? this.configContents.sourceRegions : [];
    this.inventoryPrefix = this.configContents.inventoryPrefix ? this.configContents.inventoryPrefix : 'inventory/';
    this.inventories = this.configContents.inventories;
  }
}
