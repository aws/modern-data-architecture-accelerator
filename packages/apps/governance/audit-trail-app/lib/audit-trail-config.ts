/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { CaefAppConfigParser, CaefAppConfigParserProps, CaefBaseConfigContents } from '@aws-caef/app';
import { AuditTrailProps } from '@aws-caef/audit-trail-l3-construct';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';
import * as configSchema from './config-schema.json';

export interface AuditTrailConfigContents extends CaefBaseConfigContents {
    readonly trail: AuditTrailProps
}

export class AuditTrailConfigParser extends CaefAppConfigParser<AuditTrailConfigContents> {

    public readonly trail: AuditTrailProps

    constructor( stack: Stack, props: CaefAppConfigParserProps ) {
        super( stack, props, configSchema as Schema )
        this.trail = this.configContents.trail
    }

}

