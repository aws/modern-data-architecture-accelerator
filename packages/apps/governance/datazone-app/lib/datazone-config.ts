/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParser, MdaaAppConfigParserProps, MdaaBaseConfigContents } from '@aws-mdaa/app';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';

import * as configSchema from './config-schema.json';
import { NamedDomainsProps } from '@aws-mdaa/datazone-l3-construct';

export interface DataZoneConfigContents extends MdaaBaseConfigContents {
  readonly domains: NamedDomainsProps;
}

export class DataZoneConfigParser extends MdaaAppConfigParser<DataZoneConfigContents> {
  public domains: NamedDomainsProps;
  constructor(stack: Stack, props: MdaaAppConfigParserProps) {
    super(stack, props, configSchema as Schema);
    this.domains = this.configContents.domains;
  }
}
