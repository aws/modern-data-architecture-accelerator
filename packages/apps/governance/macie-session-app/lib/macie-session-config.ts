/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MdaaAppConfigParser, MdaaAppConfigParserProps, MdaaBaseConfigContents } from '@aws-mdaa/app';
import { MacieSessionProps } from '@aws-mdaa/macie-session-l3-construct';
import { Schema } from 'ajv';
import { Stack } from 'aws-cdk-lib';

import * as configSchema from './config-schema.json';

export interface MacieSessionConfigContents extends MdaaBaseConfigContents {
  readonly session: MacieSessionProps;
}

export class MacieSessionConfigParser extends MdaaAppConfigParser<MacieSessionConfigContents> {
  public readonly macieSessionConfig: MacieSessionProps;

  constructor(stack: Stack, props: MdaaAppConfigParserProps) {
    super(stack, props, configSchema as Schema);
    this.macieSessionConfig = this.configContents.session;
  }
}
