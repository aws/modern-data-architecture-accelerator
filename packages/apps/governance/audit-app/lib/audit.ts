/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { AuditL3Construct, AuditL3ConstructProps } from '@aws-mdaa/audit-l3-construct';
import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { AuditConfigParser } from './audit-config';

export class AuditCDKApp extends MdaaCdkApp {
  constructor(props: AppProps = {}) {
    super(props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`));
  }
  protected subGenerateResources(
    stack: Stack,
    l3ConstructProps: MdaaL3ConstructProps,
    parserProps: MdaaAppConfigParserProps,
  ) {
    const appConfig = new AuditConfigParser(stack, parserProps);
    const constructProps: AuditL3ConstructProps = {
      ...{
        sourceAccounts: appConfig.sourceAccounts,
        sourceRegions: appConfig.sourceRegions,
        readRoleRefs: appConfig.readRoleRefs,
        bucketInventories: appConfig.inventories,
        inventoryPrefix: appConfig.inventoryPrefix,
      },
      ...l3ConstructProps,
    };

    new AuditL3Construct(stack, 'audit', constructProps);
    return [stack];
  }
}
