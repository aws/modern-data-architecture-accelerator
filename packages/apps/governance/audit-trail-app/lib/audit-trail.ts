/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { AuditTrailL3Construct, AuditTrailL3ConstructProps } from '@aws-mdaa/audit-trail-l3-construct';
import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { AuditTrailConfigParser } from './audit-trail-config';

export class AuditTrailCDKApp extends MdaaCdkApp {
  constructor(props: AppProps = {}) {
    super(props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`));
  }
  protected subGenerateResources(
    stack: Stack,
    l3ConstructProps: MdaaL3ConstructProps,
    parserProps: MdaaAppConfigParserProps,
  ) {
    const appConfig = new AuditTrailConfigParser(stack, parserProps);
    const constructProps: AuditTrailL3ConstructProps = {
      trail: appConfig.trail,
      ...l3ConstructProps,
    };

    new AuditTrailL3Construct(stack, 'audit-trail', constructProps);
    return [stack];
  }
}
