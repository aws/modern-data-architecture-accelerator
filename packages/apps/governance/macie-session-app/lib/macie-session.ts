/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MacieSessionL3Construct, MacieSessionL3ConstructProps } from '@aws-mdaa/macie-session-l3-construct';
import { MdaaAppConfigParserProps, MdaaCdkApp } from '@aws-mdaa/app';
import { MdaaL3ConstructProps } from '@aws-mdaa/l3-construct';
import { AppProps, Stack } from 'aws-cdk-lib';
import { MacieSessionConfigParser } from './macie-session-config';


export class MacieSessionCDKApp extends MdaaCdkApp {
    constructor( props: AppProps = {} ) {
        super( props, MdaaCdkApp.parsePackageJson(`${__dirname}/../package.json`) )
    }
    protected subGenerateResources ( stack: Stack, l3ConstructProps: MdaaL3ConstructProps, parserProps: MdaaAppConfigParserProps ) {

        const appConfig = new MacieSessionConfigParser( stack, parserProps )
        const constructProps: MacieSessionL3ConstructProps = {
            ...{
                session: appConfig.macieSessionConfig
            }, ...l3ConstructProps
        }

        new MacieSessionL3Construct( stack, "macie-session", constructProps );
        return [ stack ]

    }
}

