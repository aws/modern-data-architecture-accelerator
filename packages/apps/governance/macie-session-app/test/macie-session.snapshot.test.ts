/*!
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { describe } from '@jest/globals';
import { snapShotTest, snapShotTestApp, Create } from '@aws-mdaa/testing';
import { MacieSessionCDKApp } from '../lib/macie-session';
import * as path from 'path';

describe('macie-session Snapshot Tests', () => {
  beforeAll(() => {
    expect.addSnapshotSerializer({
      test: (val: unknown) => typeof val === 'string' && val.includes('[CONFIG:') && val.includes('test-config.yaml]'),
      print: (val: unknown) => {
        const stringVal = val as string;
        return `"${stringVal.replace(/\[CONFIG:[^[\]]*test-config\.yaml\]/, '[CONFIG:test-config.yaml]')}"`;
      },
    });
  });
  snapShotTest(
    'Macie Stack',
    Create.stackProvider(
      'MacieStackMain',
      (_, context) => {
        const moduleApp = new MacieSessionCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        return moduleApp.generateStack();
      },
      {
        module_name: 'test-macie-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );

  snapShotTestApp(
    'Macie App',
    Create.appProvider(
      context => {
        const moduleApp = new MacieSessionCDKApp({
          context: {
            ...context,
            module_configs: path.join(__dirname, 'test-config.yaml'),
          },
        });
        moduleApp.generateStack();
        return moduleApp;
      },
      {
        module_name: 'test-macie-main',
        org: 'test-org',
        env: 'test-env',
        domain: 'test-domain',
      },
    ),
  );
});
