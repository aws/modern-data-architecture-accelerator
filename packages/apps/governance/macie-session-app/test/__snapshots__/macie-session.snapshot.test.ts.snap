// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macie-session Snapshot Tests Macie App App Snapshot Test 1`] = `
{
  "test-org-test-env-test-domain-test-macie-main": {
    "Description": "(SO0320-macie-session) Modern Data Architecture Accelerator. Version 1.1.0",
    "Parameters": {
      "BootstrapVersion": {
        "Default": "/cdk-bootstrap/hnb659fds/version",
        "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
      "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
        "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
        "Type": "AWS::SSM::Parameter::Value<String>",
      },
    },
    "Resources": {
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "Description": "Stack description parameter to update on version changes",
          "Name": "/test-org/test-domain/test-macie-main/aws-solution",
          "Tags": {
            "mdaa_cdk_app": "macie-session",
            "mdaa_domain": "test-domain",
            "mdaa_env": "test-env",
            "mdaa_module_name": "test-macie-main",
            "mdaa_org": "test-org",
          },
          "Type": "String",
          "Value": "(SO0320-macie-session) Modern Data Architecture Accelerator. Version 1.1.0",
        },
        "Type": "AWS::SSM::Parameter",
      },
      "REPLACED-CFN-LOGICAL-ID": {
        "Properties": {
          "FindingPublishingFrequency": "SIX_HOURS",
          "Status": "ENABLED",
        },
        "Type": "AWS::Macie::Session",
      },
    },
    "Rules": {
      "CheckBootstrapVersion": {
        "Assertions": [
          {
            "Assert": {
              "Fn::Not": [
                {
                  "Fn::Contains": [
                    [
                      "1",
                      "2",
                      "3",
                      "4",
                      "5",
                    ],
                    {
                      "Ref": "BootstrapVersion",
                    },
                  ],
                },
              ],
            },
            "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
          },
        ],
      },
    },
  },
}
`;

exports[`macie-session Snapshot Tests Macie Stack Snapshot Test 1`] = `
{
  "Description": "(SO0320-macie-session) Modern Data Architecture Accelerator. Version 1.1.0",
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "SsmParameterValuetestorgtestdomaincaefbootstraprolehelperservicetokenC96584B6F00A464EAD1953AFF4B05118Parameter": {
      "Default": "/test-org/test-domain/caef-bootstrap/role-helper-service-token",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "Description": "Stack description parameter to update on version changes",
        "Name": "/test-org/test-domain/test-macie-main/aws-solution",
        "Tags": {
          "mdaa_cdk_app": "macie-session",
          "mdaa_domain": "test-domain",
          "mdaa_env": "test-env",
          "mdaa_module_name": "test-macie-main",
          "mdaa_org": "test-org",
        },
        "Type": "String",
        "Value": "(SO0320-macie-session) Modern Data Architecture Accelerator. Version 1.1.0",
      },
      "Type": "AWS::SSM::Parameter",
    },
    "REPLACED-CFN-LOGICAL-ID": {
      "Properties": {
        "FindingPublishingFrequency": "SIX_HOURS",
        "Status": "ENABLED",
      },
      "Type": "AWS::Macie::Session",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
